pkg/controllers/cluster/status.go

```diff
@@ -57,8 +57,9 @@ func (cc *ClusterReconciler) updateStatus(ctx context.Context, cluster *scyllav1
 		}
 		// Get corresponding StatefulSet from lister
 		err := cc.Get(ctx, naming.NamespacedName(naming.StatefulSetNameForRack(rack, cluster), cluster.Namespace), sts)
-		// If it wasn't found, continue
+		// If it wasn't found, clear rack status and continue
 		if apierrors.IsNotFound(err) {
+			delete(cluster.Status.Racks, rack.Name)
```

Comment:
This kind of shows how our current reconciliation is edge triggered instead of level triggered.

e.g. if there would be a new statefulset created in the meantime it could miss the event but I don't thing we can do much better before we go to level based reconciliation with hashes

Reply:
Agree, current approach is very error prone.